std/conv.d

```diff
@@ -1241,49 +1238,35 @@ body
 
         do
         {
-            static if (radix == 0)
-            {
-                div = cast(S)(mValue / runtimeRadix );
-                mod = cast(ubyte)(mValue % runtimeRadix);
-                mod += mod < 10 ? '0' : baseChar - 10;
-            }
-            else static if (radix > 10)
-            {
-                div = cast(S)(mValue / radix );
-                mod = cast(ubyte)(mValue % radix);
-                mod += mod < 10 ? '0' : baseChar - 10;
-            }
-            else
-            {
-                div = cast(S)(mValue / radix);
-                mod = mValue % radix + '0';
-            }
+            div = cast(S)(mValue / runtimeRadix );
+            mod = cast(ubyte)(mValue % runtimeRadix);
+            mod += mod < 10 ? '0' : baseChar - 10;
             buffer[--index] = cast(char)mod;
             mValue = div;
         } while (mValue);
 
-        static if (neg)
-        {
-            buffer[--index] = '-';
-        }
         return cast(T)buffer[index .. $].dup;
     }
 
+    import std.array;
     switch(radix)
     {
         case 10:
-            if (value < 0)
-                return toStringRadixConvert!(S.sizeof * 3 + 1, 10, true)();
-            else
-                return toStringRadixConvert!(S.sizeof * 3, 10)();
+            // The (value+0) is so integral promotions happen to the type
+            return toChars!(10, EEType)(value + 0).array;
         case 16:
-            return toStringRadixConvert!(S.sizeof * 2, 16)();
+            // The unsigned(unsigned(value)+0) is so unsigned integral promotions happen to the type
+            if (letterCase == letterCase.upper)
+                return toChars!(16, EEType, LetterCase.upper)(unsigned(unsigned(value) + 0)).array;
```

Comment:
what's `unsigned(unsigned(x)+0)` supposed to mean - provoking  integral promotion?